IonHashDotNet/IonHashWriter.cs

```
@@ -0,0 +1,342 @@
+            }
+
+            if (reader.CurrentIsNull)
+            {
+                this.WriteNull(type);
+            }
+            else
+            {
+                switch (type)
+                {
+                    case IonType.Bool:
+                        this.WriteBool(reader.BoolValue());
+                        break;
+                    case IonType.Int:
+                        this.WriteInt(reader.BigIntegerValue());
+                        break;
+                    case IonType.Float:
+                        this.WriteFloat(reader.DoubleValue());
+                        break;
+                    case IonType.Decimal:
+                        this.WriteDecimal(reader.DecimalValue());
+                        break;
+                    case IonType.Timestamp:
+                        this.WriteTimestamp(reader.TimestampValue());
+                        break;
+                    case IonType.Symbol:
+                        this.WriteSymbol(reader.StringValue());
+                        break;
+                    case IonType.String:
+                        this.WriteString(reader.StringValue());
+                        break;
+                    case IonType.Clob:
+                        this.WriteClob(reader.NewByteArray());
+                        break;
+                    case IonType.Blob:
+                        this.WriteBlob(reader.NewByteArray());
+                        break;
+                    case IonType.List:
+                        this.StepIn(IonType.List);
+                        break;
+                    case IonType.Sexp:
+                        this.StepIn(IonType.Sexp);
+                        break;
+                    case IonType.Struct:
+                        this.StepIn(IonType.Struct);
+                        break;
+                    default:
+                        throw new SystemException("Unexpected type '" + type + "'");
```

ditto:  should we throw a different exception?

---

Changed to IonHashException

---

see separate comments that propose exception types to use